service/controller/v21/templates/cloudformation/guest/vpc.go

```diff
@@ -23,4 +23,26 @@ const VPC = `{{define "vpc"}}
       Tags:
         - Key: Name
           Value: {{ $v.ClusterID }}
+  VPCS3Endpoint:
+    Type: 'AWS::EC2::VPCEndpoint'
+    Properties:
+      VpcId: !Ref VPC
+      RouteTableIds:
+        {{- range $v.RouteTableNames }}
+        - !Ref {{ .ResourceName }}
+        {{- end}}
+      ServiceName: 'com.amazonaws.{{ $v.Region }}.s3'
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: "{{ $v.ClusterID }}-vpc-s3-endpoint-policy-bucket"
+            Principal : "*"
+            Effect: "Allow"
+            Action: "s3:*"
+            Resource: "arn:{{ $v.RegionARN }}:s3:::*"
+          - Sid: "{{ $v.ClusterID }}-vpc-s3-endpoint-policy-object"
+            Principal : "*"
+            Effect: "Allow"
+            Action: "s3:*"
+            Resource: "arn:{{ $v.RegionARN }}:s3:::*/*"
```

Comment:
What is the actual difference between those two statements besides `Sid`?

Reply:
One applies on buckets, the other one applies to objects inside the buckets.

See for instance actions like `s3:GetObject`. This must apply to objects of the bucket. While action like `s3:ListBucket` applies to the  bucket itself